demo/admin/src/products/ManufacturerForm.tsx

```diff
@@ -0,0 +1,376 @@
+                                country: data.manufacturer.address.country,
+                                alternativeAddress: data.manufacturer.address.alternativeAddress
+                                    ? {
+                                          street: data.manufacturer.address.alternativeAddress.street,
+                                          streetNumber: data.manufacturer.address.alternativeAddress.streetNumber
+                                              ? String(data.manufacturer.address.alternativeAddress.streetNumber)
+                                              : null,
+                                          zip: String(data.manufacturer.address.alternativeAddress.zip),
+                                          country: data.manufacturer.address.alternativeAddress.country,
+                                      }
+                                    : undefined,
+                            }
+                          : undefined,
+                      addressAsEmbeddable: {
+                          street: data.manufacturer.addressAsEmbeddable.street,
+                          streetNumber: data.manufacturer.addressAsEmbeddable.streetNumber
+                              ? String(data.manufacturer.addressAsEmbeddable.streetNumber)
+                              : null,
+                          zip: String(data.manufacturer.addressAsEmbeddable.zip),
+                          country: data.manufacturer.addressAsEmbeddable.country,
+                          alternativeAddress: {
+                              street: data.manufacturer.addressAsEmbeddable.alternativeAddress.street,
+                              streetNumber: data.manufacturer.addressAsEmbeddable.alternativeAddress.streetNumber
+                                  ? String(data.manufacturer.addressAsEmbeddable.alternativeAddress.streetNumber)
+                                  : null,
+                              zip: String(data.manufacturer.addressAsEmbeddable.alternativeAddress.zip),
+                              country: data.manufacturer.addressAsEmbeddable.alternativeAddress.country,
+                          },
+                      },
+                  }
+                : {},
+        [data],
+    );
+
+    const saveConflict = useFormSaveConflict({
+        checkConflict: async () => {
+            const updatedAt = await queryUpdatedAt(client, "manufacturer", id);
+            return resolveHasSaveConflict(data?.manufacturer.updatedAt, updatedAt);
+        },
+        formApiRef,
+        loadLatestVersion: async () => {
+            await refetch();
+        },
+    });
+
+    const handleSubmit = async (formValues: FormValues, form: FormApi<FormValues>, event: FinalFormSubmitEvent) => {
+        if (await saveConflict.checkForConflicts()) throw new Error("Conflicts detected");
+        const output = {
+            ...formValues,
+            address: {
+                street: formValues.address.street,
```

Comment:
you should also use spread operator in nested objects for all text fields

Reply:
I now used spread-operator but this results in a graphql error 
`Variable "$input" got invalid value { __typename: "AlternativeAddress", street: "asdflj", streetNumber: null, zip: 239, country: "asdlfkj" } at "input.address.alternativeAddress"; Field "__typename" is not defined by type "AlternativeAddressInput".`

I'll search for a project doing this in a correct way...

Reply:
this is related to https://github.com/vivid-planet/comet/pull/1751/files#r1502524134